crates/sui-tool/src/db_tool/index_search.rs

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::anyhow;
+use serde::{de::DeserializeOwned, Serialize};
+use std::{path::PathBuf, str::FromStr};
+use sui_types::digests::TransactionDigest;
+use typed_store::rocks::{DBMap, MetricConf};
+use typed_store::traits::Map;
+
+use crate::get_db_entries;
+use std::fmt::Debug;
+use sui_storage::IndexStoreTables;
+use sui_types::{
+    base_types::{ObjectID, SuiAddress, TxSequenceNumber},
+    Identifier, TypeTag,
+};
+
+#[derive(Clone, Debug)]
+pub enum SearchRange<T: Serialize + Clone + Debug> {
+    ExclusiveLastKey(T),
+    Count(u64),
+}
+
+impl<T: Serialize + Clone + Debug + FromStr> FromStr for SearchRange<T>
+where
+    <T as std::str::FromStr>::Err: std::fmt::Debug,
+{
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let last_key = T::from_str(s).map_err(|e| anyhow!("Failed to parse last_key: {:?}", e))?;
+        Ok(SearchRange::ExclusiveLastKey(last_key))
+    }
+}
+
+pub fn search_index(
+    db_path: PathBuf,
```

Comment:
Maybe worth to put a note here to also update the tool whenever a new table gets created on the FN. Could be useful in a readme